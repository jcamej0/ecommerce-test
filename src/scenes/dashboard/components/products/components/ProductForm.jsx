import React, { useState } from 'react';

const ProductForm = () => {
    const [formValues, setFormValues] = useState({
        name: '',
        price: '',
        category: '',
    });

    const changeName = (event) => {
        const newName = event.target.value;
        setFormValues({...formValues, name: newName}); 
    };

    const changePrice = (event) => {
        const newPrice = event.target.value;
        setFormValues({...formValues, price: newPrice});

    };

    const changeCategory = (event) => {
        const newCategory = event.target.value;
        setFormValues({...formValues, category: newCategory});
    };

    return(
        <div>
            <h1>Nuevo producto</h1>
            <form>
                <input type="text" name="name" id="name" value={formValues.name} onChange={changeName}/>
                <label for="name">Nombre producto: </label>
                <input type="text" name="price" id="price" value={formValues.price} onChange={changePrice}/>
                <label for="price">Precio: </label>
                <select name="category" id="category" value={formValues.category} onChange={changeCategory}>
                    <option>Ropa</option>
                    <option>Joyeria</option>
                    <option>Electrodomesticos</option>
                </select>
                <label for="category">Categoria: </label>
            </form>
        </div>
    )
};

export default ProductForm;