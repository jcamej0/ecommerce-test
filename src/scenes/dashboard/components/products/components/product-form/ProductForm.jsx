import React, { useState } from "react";
import theme from "./theme.module.scss";

const ProductForm = ({ product, onSubmit }) => {
  const [formValues, setFormValues] = useState({
    name: product ? product.name : "",
    price: product ? product.price : "",
    category: product ? product.category : "",
  });
  const [ errorMessage, setErrorMessage ] = useState('');
  
  const changeName = event => {
    const newName = event.target.value;
    setFormValues({ ...formValues, name: newName });
  };

  const changePrice = event => {
    const newPrice = event.target.value;
    const regex = new RegExp("^[0-9]*$");
    
    if(!regex.exec(newPrice)) {
      return;
    }
    setFormValues({ ...formValues, price: newPrice });
  };

  const changeCategory = event => {
    const newCategory = event.target.value;
    setFormValues({ ...formValues, category: newCategory });
  };

  const checkFormError = () => {
    const {
      name,
      price,
      category,
    } = formValues;

    if(!name || !price || !category) {
      setErrorMessage("Debes llenar todos los campos");
      return false;  
    }

    return true;
  }

  const handleSubmit = event => {
      event.preventDefault();
      const noError = checkFormError();

      if(noError) {
        onSubmit(formValues);
      }
  }

  return (
    <div className={theme.container}>
      <form onSubmit={handleSubmit}>
        <h1 className={`${theme["form-title"]} ${product ? theme.edit : theme.create}`}>
          { product ? "Editar producto" : "Nuevo producto"}
        </h1>
        <label htmlFor="name">Nombre producto: </label>
        <input
          type="text"
          name="name"
          id="name"
          maxLength="20"
          value={formValues.name}
          onChange={changeName}
        />
        <label htmlFor="price">Precio: </label>
        <input
          type="text"
          name="price"
          id="price"
          maxLength="8"
          value={formValues.price}
          onChange={changePrice}
        />
        <label htmlFor="category">Categoria: </label>
        <select
          name="category"
          id="category"
          value={formValues.category}
          onChange={changeCategory}
        >
          <option defaultChecked>Electrodomesticos</option>
          <option>Patinetes</option>
          <option>Cocina</option>
          <option>Aspiraci√≥n</option>
          <option>Cuidado Personal</option>
        </select>
        <div className={theme.rule}></div>
        <div className={theme['form-footer']}>
          <button type="submit" className={`${theme.submit} ${product ? theme.edit : theme.create}`}>
            Agregar producto
          </button>
          { errorMessage &&
            <p className={theme['error-message']}>{errorMessage}</p>
          }
        </div>
      </form>
    </div>
  );
};

export default ProductForm;
