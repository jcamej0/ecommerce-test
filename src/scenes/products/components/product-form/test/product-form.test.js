import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ProductForm from '../ProductForm';


test('product form render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<ProductForm />);
		const result = renderer.getRenderOutput();
		expect(result).toMatchSnapshot();
});