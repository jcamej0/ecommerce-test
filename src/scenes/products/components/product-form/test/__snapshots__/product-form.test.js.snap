// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="container"
>
  <form
    onSubmit={[Function]}
  >
    <h1
      className="form-title create"
    >
      Nuevo producto
    </h1>
    <label
      htmlFor="name"
    >
      Nombre producto: 
    </label>
    <input
      id="name"
      maxLength="20"
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="price"
    >
      Precio: 
    </label>
    <input
      id="price"
      maxLength="8"
      name="price"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="category"
    >
      Categoria: 
    </label>
    <select
      id="category"
      name="category"
      onChange={[Function]}
      value="Electrodomesticos"
    >
      <option
        defaultChecked={true}
      >
        Electrodomesticos
      </option>
      <option>
        Patinetes
      </option>
      <option>
        Cocina
      </option>
      <option>
        Aspiración
      </option>
      <option>
        Cuidado Personal
      </option>
    </select>
    <div
      className="rule"
    />
    <div
      className="form-footer"
    >
      <button
        className="button button-green submit"
      >
        <span>
          Agregar producto
        </span>
      </button>
      
    </div>
  </form>
</div>
`;

exports[`product form render correctly 1`] = `
<div
  className="container"
>
  <form
    onSubmit={[Function]}
  >
    <h1
      className="form-title create"
    >
      Nuevo producto
    </h1>
    <label
      htmlFor="name"
    >
      Nombre producto: 
    </label>
    <input
      id="name"
      maxLength="20"
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="price"
    >
      Precio: 
    </label>
    <input
      id="price"
      maxLength="8"
      name="price"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="category"
    >
      Categoria: 
    </label>
    <select
      id="category"
      name="category"
      onChange={[Function]}
      value="Electrodomesticos"
    >
      <option
        defaultChecked={true}
      >
        Electrodomesticos
      </option>
      <option>
        Patinetes
      </option>
      <option>
        Cocina
      </option>
      <option>
        Aspiración
      </option>
      <option>
        Cuidado Personal
      </option>
    </select>
    <div
      className="rule"
    />
    <div
      className="form-footer"
    >
      <Button
        className="submit"
        color="green"
        type="submit"
      >
        Agregar producto
      </Button>
      
    </div>
  </form>
</div>
`;
